game-frontend/src/components/unitstatus/index.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { useEffect, useState } from 'react';
+import './index.css';
+
+type CircleProps = {
+  size: number;
+  progress: number;
+  strokeWidth: number;
+  circleOneStroke: string;
+  circleTwoStroke: string;
+};
+
+const UnitHealthStatus = (props: CircleProps) => {
+  // offset -> is the property , setOffset -> method to update the property
+  const [offset, setOffset] = useState(0);
+  const {
+    size,
+    progress,
+    strokeWidth,
+    circleOneStroke,
+    circleTwoStroke,
+  } = props;
+
+  const center = size / 2;
+  const radius = size / 2 - strokeWidth / 2;
+  const circumference = 2 * Math.PI * radius;
+
+  useEffect(() => {
+    // progress -> what the user will give us (unit state in this case)
+    const progressOffset = ((100 - progress) / 100) * circumference;
+    setOffset(progressOffset);
+    // set the dependency to prevent infinte update (without offset, it always returns default value 0)
+  }, [setOffset, progress, circumference, offset]);
+
+  return (
+    <div>
+      <svg className="circular-char" width={size} height={size}>
```

Comment:
Why do you work with svgs instead of the canvas?

Reply:
I did some Tutorials before hand on how to create own progress bars and they used svg. I can try to replace it with canvas, but would need some help :)

Reply:
Maybe try a tutorial working with canvas. And text us when you have questions.